api-2.2/src/main/java/org/openmrs/module/coreapps/htmlformentry/EncounterDiagnosesElement.java

```diff
@@ -209,27 +222,74 @@ public void handleSubmission(FormEntrySession formEntrySession, HttpServletReque
-                diagnosis.setEncounter(formEntrySession.getEncounter());
-                diagnosis.setCertainty(certainty == Diagnosis.Certainty.CONFIRMED ? ConditionVerificationStatus.CONFIRMED : ConditionVerificationStatus.PROVISIONAL);
-                diagnosis.setRank(diagnosisOrder == Diagnosis.Order.PRIMARY ? 1 : 2);
-                diagnosis.setPatient(formEntrySession.getPatient());
-                diagnosis.setDiagnosis(new CodedOrFreeText(answer.getCodedAnswer(), answer.getSpecificCodedAnswer(), answer.getNonCodedAnswer()));
+                if (node.path("existingDiagnosis").getNumberValue() != null) {
+                    existingDiagnosis = node.get("existingDiagnosis").getIntValue();
+                }
+
+                org.openmrs.Diagnosis diagnosis;
+
+                Integer orderId = diagnosisOrder == Diagnosis.Order.PRIMARY ? 1 : 2;
+                ConditionVerificationStatus certaintyStatus = certainty == Diagnosis.Certainty.CONFIRMED ? ConditionVerificationStatus.CONFIRMED : ConditionVerificationStatus.PROVISIONAL;
 
-                diagnosis.getEncounter().setEncounterType(Context.getEncounterService().getEncounterTypeByUuid("d7151f82-c1f3-4152-a605-2f9ea7414a79"));
-                Context.getEncounterService().saveEncounter(diagnosis.getEncounter());
-                Context.getDiagnosisService().save(diagnosis);
+                if(existingDiagnosis !=null){
+
+                    diagnosis = Context.getDiagnosisService().getDiagnosis(existingDiagnosis);
+
+                    // Check if an existing diagnosis is being resubmitted
+                    if (!existingDiagnoses.add(diagnosis)) {
+                        resubmittedDiagnoses.add(diagnosis);
+                    }
+
+                    if(!diagnosis.getRank().equals(orderId) || !diagnosis.getCertainty().equals(certaintyStatus)){
+                        diagnosis.setRank(orderId);
+                        diagnosis.setCertainty(certaintyStatus);
+                        diagnosis.setDiagnosis(diagnosis.getDiagnosis());
+                        diagnosis.setDateChanged(new Date());
+                        diagnosis.setChangedBy(Context.getAuthenticatedUser());
+                        Context.getDiagnosisService().save(diagnosis);
+                    }
+
+                }else{
+                    diagnosis = new org.openmrs.Diagnosis();
+                    diagnosis.setDiagnosis(new CodedOrFreeText(answer.getCodedAnswer(), answer.getSpecificCodedAnswer(), answer.getNonCodedAnswer()));
+                    diagnosis.setEncounter(formEntrySession.getEncounter());
+                    diagnosis.setCertainty(certaintyStatus);
+                    diagnosis.setRank(orderId);
+                    diagnosis.setPatient(formEntrySession.getPatient());
+                    diagnosis.getEncounter().setEncounterType(Context.getEncounterService().getEncounterTypeByUuid("d7151f82-c1f3-4152-a605-2f9ea7414a79"));
```

Comment:
Can we use a constant for the uuid? Something like VISIT_NOTE_ENCOUNTER_TYPE_UUID